custom_components/bosch_alarm/config_flow.py

```diff
@@ -82,3 +92,40 @@ async def async_step_user(
         return self.async_show_form(
             step_id="user", data_schema=STEP_USER_DATA_SCHEMA, errors=errors
         )
+class OptionsFlowHandler(config_entries.OptionsFlow):
+    def __init__(self, config_entry: config_entries.ConfigEntry) -> None:
+        """Initialize options flow."""
+        self.config_entry = config_entry
+
+    async def async_step_init(
+        self, user_input: dict[str, Any] | None = None
+    ) -> FlowResult:
+        """Manage the options."""
+        if user_input is not None:
+            return self.async_create_entry(title="", data=user_input)
+
+        return self.async_show_form(
+            step_id="init",
+            data_schema=vol.Schema(
+                {
+                    vol.Required(
+                        CONF_REQUIRE_ARMING_CODE,
+                        default=self.config_entry.options.get(CONF_REQUIRE_ARMING_CODE) or False,
+                    ): bool,
+
+                    vol.Required(
+                        CONF_ARMING_CODE,
+                        default=self.config_entry.options.get(CONF_ARMING_CODE) or 0,
+                    ): int,
+
+                    vol.Required(
+                        CONF_HISTORY,
+                        default=self.config_entry.options.get(CONF_HISTORY) or False,
+                    ): bool,
+                    vol.Required(
```

Comment:
Similar to the above, let's make this optional, and rename it to something like history _limit_ s.t. it makes sense that not specifying a value at all means that the fetching is unlimited, and we don't have to treat 0 as special.

Alternatively, we could just get rid of this entirely. When is it actually useful to limit the history size?

Reply:
Had someone request that, as they had a lot of history events on their panel, so they didnt want to pull all of them as it would take a while for the integration to start. The other option of course, is that we instead change it so that we append to history instead of reading it every time, which probably makes a lot more sense anyways.